common/src/main/java/com/linecorp/centraldogma/common/jsonpatch/DualPathOperation.java

```diff
@@ -60,15 +63,23 @@ abstract class DualPathOperation extends JsonPatchOperation {
      */
     DualPathOperation(final String op, final JsonPointer from, final JsonPointer path) {
         super(op, path);
-        this.from = from;
+        this.from = requireNonNull(from, "from");
+    }
+
+    /**
+     * Returns the source path.
+     */
+    public JsonPointer from() {
```

Comment:
```suggestion
    JsonPointer from() {
```
because DualPathOperation is package-private

Reply:
I intentionally made it public so users can read the value from subclasses.